weave

```diff
@@ -641,16 +641,45 @@ dns_args() {
     docker_bridge_ip
     DNS_ARG="--dns $DOCKER_BRIDGE_IP"
     DNS_SEARCH_ARG="--dns-search=."
-    for arg in "$@" ; do
-        case $arg in
+    HOSTNAME_ARG=""
+    HOSTNAME_SPECIFIED=false
+    while [ $# -gt 0 ] ; do
+        case "$1" in
             --dns-search=*)
                 DNS_SEARCH_ARG=""
+                shift
+                ;;
+            --name)
+                HOSTNAME_ARG="--hostname=$2.$(dns_domain)"
+                shift 2
+                ;;
+            --name=*)
+                HOSTNAME_ARG="--hostname=${1#*=}.$(dns_domain)"
```

Comment:
I suggest deferring the `$(dns_domain)`, so we don't execute it unnecessarily when `$HOSTNAME_SPECIFIED` turns out to be true.